### PR TITLE
Added Route Protection

### PR DESCRIPTION
**Pull request by AkshatSharma5**

### CODE REVIEW

File: package.json
Comment (AryanGKulkarni) [inline package.json:1]: No need of this lib

use this
``` import Cookies from "js-cookie";```

File: package.json
Comment (AryanGKulkarni) [inline package.json:34]: No need to use this lib

instead use this
```import Cookies from "js-cookie";```

File: src/components/Sidebar.tsx
Comment (AryanGKulkarni) [inline src/components/Sidebar.tsx:84]: I hope this was used for testing, remove the changes made in this file

File: src/middleware.ts
Comment (AryanGKulkarni) [inline src/middleware.ts:24]: write this
```return NextResponse.redirect(new URL("/login", request.url));```

File: src/components/Sidebar.tsx
Comment (AryanGKulkarni) [inline src/components/Sidebar.tsx:84]: I hope this was used for testing, revert these changes

File: src/middleware.ts
Comment (AryanGKulkarni) [inline src/middleware.ts:7]: why are these lines commented?

File: src/middleware.ts
Comment (AryanGKulkarni) [inline src/middleware.ts:17]: Instead of this write

```if(isAdmin && !(request.url.includes('/admin')))```

File: src/middleware.ts
Comment (AryanGKulkarni) [inline src/middleware.ts:23]: same

File: src/middleware.ts
Comment (AkshatSharma5) [reply src/middleware.ts:7]: they were used for testing.

File: src/middleware.ts
Comment (AkshatSharma5) [reply src/middleware.ts:17]: fixed

File: src/middleware.ts
Comment (AkshatSharma5) [reply src/middleware.ts:20]: fixed

File: src/middleware.ts
Comment (AkshatSharma5) [reply src/middleware.ts:23]: fixed

**Comment by mittal-ishaan** [comment]
merge this after @Pradeep-Kumar-Rebbavarapu  merges his pr. and use yarn.

**Review by mittal-ishaan — DISMISSED** [review]

**Comment by mittal-ishaan** [comment]
only @AryanGKulkarni should merge this

**Review by AryanGKulkarni — APPROVED** [review]

**Pull request merged**
